src/axes/axis.ts

```diff
@@ -150,9 +178,266 @@ export class Axis<D> extends Component {
+      .attr({
+        x1: isHorizontal ? positionF : secondaryPosition,
+        x2: isHorizontal ? positionF : offsetF,
+        y1: isHorizontal ? secondaryPosition : positionF,
+        y2: isHorizontal ? offsetF : positionF,
+        visibility: visibilityF
+      });
+
+    bindElements(this._annotationContainer.select(".annotation-circle-container"), "circle", "annotation-circle")
+      .attr({
+        cx: isHorizontal ? positionF : secondaryPosition,
+        cy: isHorizontal ? secondaryPosition : positionF,
+        r: 3,
+        visibility: visibilityF
+      });
+
+    let rectangleOffsetF = (d: D) => {
+      switch (this.orientation()) {
+        case "bottom":
+        case "right":
+          return offsetF(d);
+        case "top":
+        case "left":
+          return offsetF(d) - measurements.get(d).height;
+      }
+    };
+    bindElements(this._annotationContainer.select(".annotation-rect-container"), "rect", "annotation-rect")
+      .attr({
+        x: isHorizontal ? positionF : rectangleOffsetF,
+        y: isHorizontal ? rectangleOffsetF : positionF,
+        width: isHorizontal ? (d) => measurements.get(d).width : (d) => measurements.get(d).height,
+        height: isHorizontal ? (d) => measurements.get(d).height : (d) => measurements.get(d).width,
+        visibility: visibilityF
+      });
+
+    let annotationWriter = this._annotationWriter;
+    let annotationFormatter = this.annotationFormatter();
+    let annotationLabels = bindElements(this._annotationContainer.select(".annotation-label-container"), "g", "annotation-label");
+    annotationLabels.selectAll(".text-container").remove();
+    annotationLabels.attr({
+        transform: (d) => {
+          let xTranslate = isHorizontal ? positionF(d) : rectangleOffsetF(d);
+          let yTranslate = isHorizontal ? rectangleOffsetF(d) : positionF(d);
+          return `translate(${xTranslate},${yTranslate})`;
+        },
+        visibility: visibilityF
+      })
+      .each(function (annotationLabel) {
+        let writeOptions = {
+          selection: d3.select(this),
+          xAlign: "center",
+          yAlign: "center",
+          textRotation: isHorizontal ? 0 : 90
+        };
+        annotationWriter.write(annotationFormatter(annotationLabel),
+                                 isHorizontal ? measurements.get(annotationLabel).width : measurements.get(annotationLabel).height,
+                                 isHorizontal ? measurements.get(annotationLabel).height : measurements.get(annotationLabel).width,
+                                 writeOptions);
+      });
+  }
+
+  private _annotatedTicksInDomain() {
+    let scaleRange = this._scale.range();
+    return this.annotatedTicks().filter((tick) => {
+      if (tick == null) {
+        return false;
+      }
+      return Utils.Math.inRange(this._scale.scale(tick), scaleRange[0], scaleRange[1]);
+    });
+  }
+
+  protected _axisSizeWithoutMarginAndAnnotations() {
```

Comment:
This name seems kind of flexible and unwieldy, since it's defined by what's *not* in it rather than what *is*. Could this method be replaced with a combination of other methods that return the height of each part of the `Axis`?